lib/sidekiq/cron/job.rb

```diff
@@ -239,12 +241,11 @@ def self.count
       def self.find name
         # If name is hash try to get name from it.
         name = name[:name] || name['name'] if name.is_a?(Hash)
+        return unless exists? name
 
         output = nil
         Sidekiq.redis do |conn|
-          if exists? name
```

Comment:
I wonder if this could cause a deadlock since `exists?` also checks out a connection.

Reply:
Probably not, since `connection_pool` appears to reuse the same connection for a given thread.